src/client/opamCliMain.ml

```diff
@@ -110,9 +110,11 @@ let check_and_run_external_commands () =
             if OpamCLIVersion.is_supported cli then
               let () =
                 if OpamCLIVersion.(cli >= (2, 1)) then
+                  let flag = "--cli=" ^ OpamCLIVersion.(to_string cli) in
                   OpamConsole.warning
-                    "Setting OPAMCLI is brittle - consider using the \
-                     '--cli <major>.<minor>' flag."
+                    "OPAMCLI should only ever be set to %s - use '%s' instead."
```

Comment:
Is this really a necessary warning? In Dockerfiles for example, it would be quite hard to weave through a choice of "--cli" or "ENV OPAMCLI=2.0" without programmatic generation. Just setting OPAMCLI=2.1 is both reliable and easy in that environment -- what's brittle about environment variables if you control the environment?

Reply:
The main reason was hygiene - you should control the _CLI_ version using the _CLI_! You never truly control the environment, or you at least need to be paranoid at _every_ call site to the opam binary that `OPAMCLI` is correct - the aim is to nudge all scripts (and all programs) to specify `--cli` with the command being executed (like a type annotation!). The secondary reason was to strongly discourage exporting `OPAMCLI` in `.bashrc`.

Reply:
For scripts, the expectation was that you'd do something like `opam='opam --cli=2.1' at the top of your script (or equivalent in better languages!) (cf. https://github.com/ocaml/opam/wiki/Spec-for-opam-CLI-versioning#summary)

Reply:
Got it, thanks. I'd thought that was a hard error, but setting OPAMCLI=2.0 is actually only a warning as well. That does make it easier to transition via env variables "in case of scripting emergency"

Reply:
Definitely - the only hard-error you should get is explicitly requesting 2.0 and trying to use a 2.1 feature or the sort of vice-versa (requesting 2.1 and trying to use something from 2.0 which has been removed/renamed from the CLI)